core/dbt/artifacts/resources/v1/snapshot.py

```diff
@@ -52,6 +52,10 @@ def validate(cls, data):
         if data.get("materialized") and data.get("materialized") != "snapshot":
             raise ValidationError("A snapshot must have a materialized value of 'snapshot'")
 
+        # Validate if there are duplicate column names in check_cols.
+        if len(data.get("check_cols")) != len(set(data.get("check_cols"))):
```

Comment:
Even though `.get()` is used, if `check_cols` is missing, this will result in an error because `NoneType has no len()`.

```suggestion
        if "check_cols" in data and len(data["check_cols"]) != len(set(data["check_cols"])):
```

Reply:
@jordivandooren Could you please review the latest changes? I updated my code, and everything is now up-to-date. The duplicated columns validation has been moved inside the final_validate function.